### PR TITLE
#170947693 users should be able to chat 

### PR DESCRIPTION
**Pull request by Niyonsengaeric**
#### What does this PR do?
- Users should be able to chat on Barefoot Nomad.
#### Description of Task to be completed?
- View all of the past chats.
- Post a public chat
- Echoed message to all online users in realtime.
#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git checkout ft-chat-on-barefoot-170947693` to switch to this branch.
- run `npm install` to install dependencies.
- create a Postgres database.
- create a .env file then add environment variable as shown below:
```
DB_USER='name of the user of the database'
DB_PASS='password of your postgress'
DB_NAME='name of the database || Barefoot'
DB_TEST_NAME='name of the database for test || Barefoot_test'
NODE_ENV='development'
SECRETKEY='PrivateKey'
sendgridKey='your-unique-sendgrid-api-key'
MY_MAIL='your-email-address' 
MY_PWORD='your-password' 
GOOGLE_CLIENT_ID = 'your-google-client-id'
GOOGLE_CLIENT_SECRET = 'your-gogle-secret'
GOOGLE_CALLBACK_URL = 'https://your-app-on-google'
FCBK_CLIENT_ID = 'your-fb-id'
FCBK_CLIENT_SECRET = 'your-fb-secret'
FCBK_CALLBACK_URL = 'https://your-heroku-app-on-facebook'
HOST_NAME='your-heroku-staging-app'
LOCAL_HOST='http://localhost:port'
```
- use .env.example as a blueprint for your .env file.
- run `npm run createTables` to create the database tables.
- run `npm dev-start` to start the server.
- open the browser use URL:`http://localhost:4000/api/v1/docs` for swagger documentation.
- Open postman and copy this URL POST `http://localhost:4000/api/v1/auth/login`. to login to the system if you have an account.
- If you don't have an account open postman and copy this URL POST `localhost:4000/api/v1/auth/signup`  to sign up
- Use this format to Sign up
```
{
	"firstName": "Kagorora",
	"lastName": "Messi",
	"gender": "Male",
	"PassportNumber": "12345675",
	"email": "example@gmail.com ",
	"password": "Niyonkuru@1"
}
```
- Use this user to login as manager
```
{
 "email": "william.ishimwe@andela.com",
 "password": "Password@1" 
}
             
```
- If a user is successfully signup or login 
#### View all chats
 - Open postman and copy and send this URL GET `http://localhost:4000/api/v1/chat`.
- You will view all of the past chats.
#### Post a public chat

 [Download socket tester](https://github.com/AppSaloon/socket.io-tester/releases/download/v1.2.3/socket-io-tester-darwin-x64.zip) 

- user `http://localhost:4000` as URL
- Open it and enter `send-chat-message` as Send Message and `send-chat-message-2` as Listen for events after, send the message in the field of the message above send message button: 
You will see the results as shown in the pictures below.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947693](https://www.pivotaltracker.com/story/show/170947693)
#### Screenshots 

                               View all of the past chats

Screen Shot 2020-03-11 at 00 13 41

                               Post a public chat

Screen Shot 2020-03-11 at 00 10 20

#### Questions:
N/A

### CODE REVIEW

**Comment by Kagorora** [comment]
@Niyonsengaeric  nice job, I can see all the charts via postman,tho I am wondering if there is an option/way to view all chart live via the socket.io.tester tool that you suggested that we downloaded, If there is a way to see those notification live as your PT requires, please show how to do it on your PR body.

**Comment by Niyonsengaeric** [comment]
> @Niyonsengaeric nice job, I can see all the charts via postman,tho I am wondering if there is an option/way to view all chart live via the socket.io.tester tool that you suggested that we downloaded, If there is a way to see those notification live as your PT requires, please show how to do it on your PR body.

@Kagorora  am not getting well you question maybe you can make it more clear

**Review by Kagorora — APPROVED** [review]
@Niyonsengaeric  you did an excellent job, and I think this PR can be merged

**Review by Moise1 — APPROVED** [review]

File: src/app.js [hunk 1]
```diff
@@ -48,12 +54,28 @@ app.use((req, res, next) => {
...
+  //     socket.on('username', async () => {
+  //       socket.username = await `${firstName} ${lastName}`;
+  //       io.emit('is_online', `${socket.username} join the chat..`);
+  //     });
+  //     if (connectedSockets.indexOf(socket) === -1) {
+  //       const username = `${firstName} ${lastName}`;
+  //       connectedSockets.push({ id, username });
+  //     }
+  //   }
+  // }
```
Comment (kagabof) [inline src/app.js]: Why do you have unused code?

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/app.js (hunk 1 above)
Comment (Niyonsengaeric) [reply src/app.js]: sorry I forget to remove it

**Review by kagabof — APPROVED** [review]
Great job.

**Pull request merged**